### DISCUSSION TITLE
Which loss could deal with unsupervised link prediction of heterogeneous graph?

### DISCUSSION THREAD

**Discussion by wanglu2014** [pyg-team/pytorch_geometric#2453] [General]
The main question is the loss function. Thank you for your help!

**Comment by rusty1s** [comment]
I guess the loss formulation for link prediction in heterogeneous graphs is the same for link prediction in homogeneous graphs: You would need to utilize a contrastive loss formulation, e.g., via the negative log likelihood of positive and negative sampled examples.

**Reply by wanglu2014** [reply]
Thank you for your timely reply. Our goal is to focus on edges between two types of nodes. Did you have implementation?

**Reply by rusty1s** [reply]
Don't have an implementation for that, sorry, but I guess you can come up with something quite quickly by looking at `examples/link_pred.py` and a heterogeneous graph model, e.g., [this one](https://github.com/snap-stanford/ogb/blob/master/examples/nodeproppred/mag/rgcn.py).
